library/src/contraction/contraction_selection.cpp

```diff
@@ -175,6 +194,25 @@ namespace hiptensor
         }
     }
 
+    bool is1D(std::vector<std::size_t> const& a_ms_ks_lengths,
+              std::vector<std::size_t> const& a_ms_ks_strides,
+              std::vector<std::size_t> const& b_ns_ks_lengths)
+    {
```

Comment:
a_ms_ks_strides argument is unused in this function.

What is important here? Any length is 1? Or stride?

Reply:
This was a remnant from a previous implementation, It's removed now.

Reply:
Might want to have an alternate name that is more suggestive, as at first glance this could lead you to assume it's checking if lengths.size() == 1.
Possible suggestion: maybe can replace this function with something more compact like std::count:

`std::count(v.cbegin(), v.cend(), 1) `